packages/clerk-js/src/core/clerk.ts

```diff
@@ -1280,13 +1282,27 @@ export default class Clerk implements ClerkInterface {
     if (!this.isSatellite) {
       return;
     }
+
     if (this.#instanceType === 'development' && !this.#options.signInUrl) {
       clerkMissingSignInUrlAsSatellite();
     }
 
     if (!this.proxyUrl && !this.domain) {
       clerkMissingProxyUrlAndDomain();
     }
+
+    if (this.#options.signInUrl) {
+      let signInUrl: URL;
+      try {
+        signInUrl = new URL(this.#options.signInUrl);
+      } catch {
+        clerkInvalidSignInUrlFormat();
+      }
+
+      if (signInUrl.hostname === this.domain.replace('clerk.', '')) {
```

Comment:
@desiprisg Should we update this condition to match the logic from the `backend` package ?